### PR TITLE
Feat: extend response api

### LINKED ISSUE #1587: feature: extend Response API
We should specify concrete broker Response classes (like [KafkaResponse](https://github.com/airtai/faststream/blob/main/faststream/kafka/response.py#L4)) to make them supporting all broker features ([Publisher.publish](https://github.com/airtai/faststream/blob/main/faststream/kafka/publisher/usecase.py#L100) options)

Another words, **KafkaResponse** should has the following signature:

```python
class KafkaResponse(Response):
    def __init__(
        self,
        body: "SendableMessage",
        *,
        headers: Optional["AnyDict"] = None,
        correlation_id: Optional[str] = None,
        # Kafka specific
        timestamp_ms: Optional[int] = None,
        key: Optional[bytes] = None,
        ...
    ) -> None:
        ...
```

Another brokers Response classes should be extended the same way

### PR DESCRIPTION
**Pull request by Flosckow**
# Description

Please include a summary of the change and specify which issue is being addressed. Additionally, provide relevant motivation and context.

Fixes #1587 

## Type of change

Please delete options that are not relevant.

- [ ] Documentation (typos, code examples, or any documentation updates)
- [ ] Bug fix (a non-breaking change that resolves an issue)
- [x] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a fix or feature that would disrupt existing functionality)
- [ ] This change requires a documentation update

## Checklist

- [x] My code adheres to the style guidelines of this project (`scripts/lint.sh` shows no errors)
- [x] I have conducted a self-review of my own code
- [ ] I have made the necessary changes to the documentation
- [ ] My changes do not generate any new warnings
- [ ] I have added tests to validate the effectiveness of my fix or the functionality of my new feature
- [x] Both new and existing unit tests pass successfully on my local environment by running `scripts/test-cov.sh`
- [x] I have ensured that static analysis tests are passing by running `scripts/static-analysis.sh`
- [ ] I have included code examples to illustrate the modifications

### CODE REVIEW

File: faststream/kafka/response.py [hunk 1]
```diff
@@ -1,5 +1,19 @@
...
 
 class KafkaResponse(Response):
-    pass
+    def __init__(
+        self,
+        body: "SendableMessage",
+        *,
+        headers: Optional["AnyDict"] = None,
+        correlation_id: Optional[str] = None,
+        partition: Optional[int] = None,
```
Comment (Lancetnik) [inline faststream/kafka/response.py:16]: Remove `partition`

File: faststream/nats/response.py [hunk 2]
```diff
@@ -1,5 +1,20 @@
...
 
 class NatsResponse(Response):
-    pass
+    def __init__(
+        self,
+        body: "SendableMessage",
+        *,
+        headers: Optional["AnyDict"] = None,
+        correlation_id: Optional[str] = None,
+        timeout: Optional[float] = None,
```
Comment (Lancetnik) [inline faststream/nats/response.py:16]: Remove `timeout` and all RPC options

File: faststream/rabbit/response.py [hunk 3]
```diff
@@ -1,5 +1,22 @@
...
+
 
 class RabbitResponse(Response):
-    pass
+    def __init__(
+        self,
+        body: "AioPikaSendableMessage",
+        *,
+        correlation_id: Optional[str] = None,
+        queue: Optional["RabbitQueue"] = None,
```
Comment (Lancetnik) [inline faststream/rabbit/response.py:16]: Remove `queue`, `routing_key` and RPC options. Add `headers`

File: faststream/redis/response.py [hunk 4]
```diff
@@ -1,5 +1,19 @@
...
 
 class RedisResponse(Response):
-    pass
+    def __init__(
+        self,
+        body: Optional["SendableMessage"] = None,
+        *,
+        headers: Optional["AnyDict"] = None,
+        correlation_id: Optional[str] = None,
+        rpc: bool = False,
```
Comment (Lancetnik) [inline faststream/redis/response.py:16]: Remove all RPC options

File: faststream/confluent/response.py [hunk 5]
```diff
@@ -1,5 +1,19 @@
...
 
 class KafkaResponse(Response):
-    pass
+    def __init__(
+        self,
+        body: "SendableMessage",
+        *,
+        headers: Optional["AnyDict"] = None,
+        correlation_id: Optional[str] = None,
+        partition: Optional[int] = None,
```
Comment (Lancetnik) [inline faststream/confluent/response.py:16]: Remove `partition` here

**Review by Lancetnik — CHANGES_REQUESTED** [review]

File: faststream/rabbit/response.py (hunk 3 above)
Comment (Lancetnik) [reply faststream/rabbit/response.py:16]: Also, you forgot about RMQ extra options: there are a lot of them - https://github.com/Flosckow/faststream/blob/feat-extend-response-api/faststream/rabbit/publisher/usecase.py#L220

**Comment by Lancetnik** [comment]
@Flosckow don't forget to initiate classes. Now you have only `__init__(): ...`, but it should use super and setup custom fields as well

**Comment by Lancetnik** [comment]
And finally, don't forget to override `Response.as_publish_kwargs` for each class

File: faststream/rabbit/response.py (hunk 3 above)
Comment (Flosckow) [reply faststream/rabbit/response.py:16]: Yes, indeed, I missed that. We are talking about Publish Kwargs, which also need to be passed to method init?

Comment (Lancetnik) [reply faststream/rabbit/response.py:16]: Yep

**Comment by Flosckow** [comment]
If we have equal argument in specific woker response class, can we not override as_publish_kwargs, or is it better to do it right away, in case something comes up later?

**Comment by Flosckow** [comment]
Can we use AioPikaSendableMessage(need add to base response class inside union), or more correctly strategy use SendableMessage inside rmq response?

**Review by Lancetnik — DISMISSED** [review]

**Review by Lancetnik — APPROVED** [review]

**Review by Lancetnik — APPROVED** [review]

**Review by Lancetnik — APPROVED** [review]

**Pull request merged**